### PR TITLE
Added compatibility with koa@2

### PR DESCRIPTION
**Pull request by bertofer**

### CODE REVIEW

File: test/simple.js [hunk 1]
```diff
@@ -4,31 +4,31 @@ var request = require('supertest');
 var should = require('should');
 var route = require('koa-route');
 var range = require('../');
-var koa = require('koa');
-var app = koa();
+var Koa = require('koa');
+var app = new Koa();
```
Comment (yorkie) [inline test/simple.js]: Remove the above 2 lines

**Comment by yorkie** [comment]
Try to bump version, too

**Comment by yorkie** [comment]
I'm not familiar with koa2, @dead-horse may I ask you to have a review as well :)

File: test/simple.js (hunk 1 above)
Comment (bertofer) [inline test/simple.js]: What do you mean? Leaving the old version? In Koa 2, the app has to be initialized with the `new` keyword.

Comment (yorkie) [inline test/simple.js]: +1

**Comment by yorkie** [comment]
@bertofer koa2 does support async/await natively?

**Comment by bertofer** [comment]
koa2 yes, the problem is with node. Right now you can:

- Use a transpiler like babel
- Use node 7 with --harmony-async-await to use the native not-stable-yet implementation of async/await.

**Comment by yorkie** [comment]
So in this PR, we should do one of the following 2 things:

1. compile the es2016 source code by babeljs when publishing npmjs, we can't require user to use something like babeljs.
2. add an "engine" line to specify this package requires node7+ and `--harmony-async-await` v8 flag.

**Comment by bertofer** [comment]
Hi @yorkie,

Sorry for the delay on this.

I wouldn't do **1.**, as this library will only be compatible with koa@2, and to use koa 2 you already need to have a transpiler in place, or either use `--harmony-async-await` with node v7.

As for **2.**, sounds good for me but I am not sure the flag `--harmony-async-await` can be added to the engine property.

In case the flag cannot be added to the engines, what about just specifying node v7 and modify the README.md to specify that you need to do any of the 2 options to get it working for koa@2?

**Comment by yorkie** [comment]
> In case the flag cannot be added to the engines, what about just specifying node v7 and modify the README.md to specify that you need to do any of the 2 options to get it working for koa@2?

That's what I did mean :)

**Comment by bertofer** [comment]
Will investigate how to change the travis build to use node 7 with `--harmony-async-await`, so the pr can be merged.

File: Makefile [hunk 2]
```diff
@@ -5,11 +5,13 @@ test:
 	@NODE_ENV=test ./node_modules/.bin/mocha \
 		--require should \
 		--harmony \
+    --harmony-async-await \
```
Comment (yorkie) [inline Makefile]: alignment with other lines :)

File: Makefile [hunk 3]
```diff
@@ -5,11 +5,13 @@ test:
...
 		test/*.js \
 		--bail
 
 test-cov:
 	@NODE_ENV=test node --harmony \
+    --harmony-async-await \
```
Comment (yorkie) [inline Makefile]: ditto

File: Makefile [hunk 4]
```diff
@@ -19,6 +21,7 @@ test-cov:
 
 test-travis:
 	node --harmony \
+    --harmony-async-await \
```
Comment (yorkie) [inline Makefile]: ditto

File: README.md [hunk 5]
```diff
@@ -18,36 +18,38 @@ range request implementation for koa
...
 // via readable stream
-app.use(route.get('/stream', function * () {
-  this.body = fs.createReadStream('your path');
+app.use(route.get('/stream', async function (ctx) {
+  ctx.body = fs.createReadStream('your path');
 }));
 
 ```
 
+Until async/await is supported by default, you will need to transpile your code with something like Babel, or use Node v7 with the `--harmony-async-await` flag.
```
Comment (yorkie) [inline README.md]: Replace by the following:

```
Until `async/await` is supported by default, you need to transpile your code by one of the following:

- Bebel
- Node7 with the `--harmony-async-await` flag.
```

**Comment by yorkie** [comment]
@bertofer We are almost done, just fix the last 2 small nit about docs :)

File: README.md (hunk 5 above)
Comment (bertofer) [inline README.md]: Changed a little bit, as using node7 is not transpiling, so could be confusing ;)

**Comment by bertofer** [comment]
Hi @yorkie , any news? :)

**Comment by yorkie** [comment]
Released at v3.0.0, thank you @bertofer :)

**Pull request merged**